webapp/src/components/admin_settings/jitsi_settings.tsx

```diff
@@ -0,0 +1,280 @@
+    constructor(props: any) {
+        super(props);
+
+        let selectedMode = JITSI_MODE;
+
+        // Check previous setting setup in case of upgrade
+
+        if (this.props.value) {
+            selectedMode = this.props.value.usejaas === true ? JAAS_MODE : JITSI_MODE;
+        }
+
+        let selectedSettings = DEFAULT_SETTINGS;
+
+        if (!this.props.value) {
+            selectedSettings = {
+                jitsiurl: this.props.config.PluginSettings.Plugins[pluginId].jitsiurl || DEFAULT_SETTINGS.jitsiurl,
+                jitsiappsecret: this.props.config.PluginSettings.Plugins[pluginId].jitsiappsecret || DEFAULT_SETTINGS.jitsiappsecret,
+                jitsiappid: this.props.config.PluginSettings.Plugins[pluginId].jitsiappid || DEFAULT_SETTINGS.jaasappid,
+                jitsicompatibilitymode: this.props.config.PluginSettings.Plugins[pluginId].jitsicompatibilitymode || DEFAULT_SETTINGS.jitsicompatibilitymode,
+                jitsiembedded: this.props.config.PluginSettings.Plugins[pluginId].jitsiembedded || DEFAULT_SETTINGS.jitsiembedded,
+                jitsijwt: this.props.config.PluginSettings.Plugins[pluginId].jitsijwt || DEFAULT_SETTINGS.jitsijwt,
+                jitsilinkvalidtime: this.props.config.PluginSettings.Plugins[pluginId].jitsilinkvalidtime || DEFAULT_SETTINGS.jitsilinkvalidtime,
+                jitsinamingscheme: this.props.config.PluginSettings.Plugins[pluginId].jitsinamingscheme || DEFAULT_SETTINGS.jitsinamingscheme,
+                usejaas: DEFAULT_SETTINGS.usejaas,
+                jaasappid: DEFAULT_SETTINGS.jaasappid,
+                jaasapikey: DEFAULT_SETTINGS.jaasapikey,
+                jaasprivatekey: DEFAULT_SETTINGS.jaasprivatekey
+            };
+        }
+
+        this.state = {
+            mode: selectedMode,
+            value: this.props.value || selectedSettings
+        };
+
+        this.onJaaSApiKeyChanged = this.onJaaSApiKeyChanged.bind(this);
+        this.onJaaSAppIDChanged = this.onJaaSAppIDChanged.bind(this);
+        this.onJaaSPrivateKeyChanged = this.onJaaSPrivateKeyChanged.bind(this);
+        this.onJaaSEmbeddedChanged = this.onJaaSEmbeddedChanged.bind(this);
+        this.onJaaSCompatibilityChange = this.onJaaSCompatibilityChange.bind(this);
+
+        this.onJitsiAppIDChanged = this.onJitsiAppIDChanged.bind(this);
+        this.onJitsiAppSecretChanged = this.onJitsiAppSecretChanged.bind(this);
+        this.onJitsiAuthChanged = this.onJitsiAuthChanged.bind(this);
+        this.onJitsiCompatibilityChanged = this.onJitsiCompatibilityChanged.bind(this);
+        this.onJitsiEmbeddedChanged = this.onJitsiEmbeddedChanged.bind(this);
+        this.onJitsiMeetingLinkExpChanged = this.onJitsiMeetingLinkExpChanged.bind(this);
+        this.onJitsiMeetingNamesChanged = this.onJitsiMeetingNamesChanged.bind(this);
+        this.onJitsiURLChanged = this.onJitsiURLChanged.bind(this);
+    }
+
+    componentDidMount() {}
```

Comment:
Any reason to define these empty functions?

Reply:
Removed it.